### ISSUE TITLE
support JSON config as well as YAML

### ISSUE THREAD

**Issue by leondz** [NVIDIA/garak#913]
## Summary
YAML isn't a universally powerful and flawless format. Consider also supporting JSON for standard configs.

### Basic example

E.g. [`fast.yaml`](https://github.com/leondz/garak/blob/main/garak/configs/fast.yaml) would become:

```
{
  "system": {
    "parallel_attempts": 20,
    "lite": true
  },
  "run": {
    "generations": 5
  },
  "plugins": {
    "probe_spec": "continuation,dan,encoding.InjectBase64,encoding.InjectHex,goodside,av_spam_scanning,leakreplay,lmrc,malwaregen.SubFunctions,malwaregen.TopLevel,packagehallucination,realtoxicityprompts.RTPIdentity_Attack,realtoxicityprompts.RTPProfanity,realtoxicityprompts.RTPSexually_Explicit,realtoxicityprompts.RTPThreat,snowball,xss",
    "extended_detectors": false
  }
}
```

### Motivation

> Why are we doing this? What use cases does it support? What is the expected outcome?

1. YAML lacks some definitional strength, making certain values difficult to express, and an alternative is useful
2. The internal config structure is agnostic to which text format the config is stored in, hopefully keeping the scope of code changes relatively limited
3. It's good to provide us with flexibility around something as core and powerful as configs
4. JSON schema validation is trivial and built-in to python, enabling strong config checking pre-run 
5. We will gain a quick, reasonable answer to the various gripes about yaml that I (and presumably also the world) have, like how the official YAML implementation in Python doesn't follow the official YAML spec

**Comment by patriciapampanelli** [comment]
@leondz Should site config and core config support JSON format? Should files without extensions be allowed?

**Comment by leondz** [comment]
Yes, I think that makes sense - it seems hard to justify excluding them

**Comment by patriciapampanelli** [comment]
Do you think it makes sense that, in cases where two files in the same folder share the same name but different extensions (`.json` and `.yaml`), we should prioritize the JSON version? I can add this clarification to the documentation if that sounds reasonable.

**Comment by leondz** [comment]
Oh, great question. I'd be inclined to prioritise JSON - @jmartin-tech do you have an opinion?

**Comment by jmartin-tech** [comment]
I would suggest automatically utilized files should stick to one format and support for others format requires explicit filename with extension.

Official project packages should not ship with ambiguous file names and `configs` are not in the data path so user override is not currently supported.  _If_ we want to support selection of files from the default XDG based config directory other than `garak.site.yaml` I would suggest we should just raise a configuration error if ambiguity is found.

**Comment by jmartin-tech** [comment]
Also on the question:
> Should site config and core config support JSON format? 

I would say `site` config should support both, however `core` is not a user configurable file it is shipped only as part of the package and for a non dev user it exists in the python package install path and should be considered a fully resolved constant.

**Comment by leondz** [comment]
resolved in #1490
